### DISCUSSION TITLE
Duplication between Equinox P2 and Mylyn

### DISCUSSION THREAD

**Discussion by ruspl-afed** [eclipse-equinox/p2#306] [General]
There is an extension point duplication between **Equinox P2** (https://github.com/eclipse-equinox/p2/blob/master/bundles/org.eclipse.equinox.p2.discovery.compatibility/plugin.xml#L16 ) and **Mylyn** (https://github.com/eclipse-mylyn/org.eclipse.mylyn/blob/main/mylyn.commons/org.eclipse.mylyn.discovery.core/plugin.xml#L13)

What should we do with it?

**Comment by merks** [comment]
I see a commit with this message:

> bug 295273 - [discovery] move Mylyn Connector Discovery UI into P2

So I guess stuff got copied from Mylyn to p2, and then was not deleted from Mylyn.  I don't know what this is for or if anyone is using it, but there are tests and a hierarchy under org.eclipse.equinox.internal.p2.discovery.AbstractDiscoveryStrategy.  

So probably the Mylyn "copy" should be deleted....

**Reply by ruspl-afed** [reply]
Yes, I also see this commit. And following the discussion at https://bugs.eclipse.org/bugs/show_bug.cgi?id=295273 I would also assume that Mylyn copy should be deleted. 
However, I wonder why P2 code still contains so many "Mylyn" words, including extension point id `org.eclipse.mylyn.discovery.core.connectorDiscovery`

**Reply by merks** [reply]
It could just be that it was just poorly done without much review.   I'm not sure who can really say at this point in time...

**Comment by laeubi** [comment]
> However, I wonder why P2 code still contains so many "Mylyn" words, including extension point id

As far as I know this is sued e.g. for SVN connector and m2e connectors as well, so if it would have been renamed meaning it breaks existing plugins.

**Reply by ruspl-afed** [reply]
> if it would have been renamed meaning it breaks existing plugins.

In other words you suggest to keep `org.eclipse.mylyn.discovery.core.connectorDiscovery` extension point inside P2 as it is now, right?

**Reply by laeubi** [reply]
I just can guess why it was not renamed, this is the wiki that explains the feature itself:

https://wiki.eclipse.org/Mylyn/Discovery
https://wiki.eclipse.org/Equinox/p2/Discovery

I only know SVN and M2e using it, maybe @fbricon can tell more about how/if m2e would be able to migrate to a new extension point.

**Reply by fbricon** [reply]
m2e used the mylyn extension point before it was moved upstream. From the top pf my head, I don't see any good reason to keep using it if we can replace it with the p2 one. But someone would have to check of course.

**Reply by merks** [reply]
I also don’t see a compelling reason to change the id 13 years afterward the migration.  What benefits would such a change bring?

**Reply by ruspl-afed** [reply]
I had no plan to change the id in P2, I was rather surprised how two extension points with the same id can coexist for so long without any troubles reported. 
OK, the full story is: I was trying to proceed with  https://github.com/eclipse-mylyn/org.eclipse.mylyn/issues/196 , then I found this duplication and it baffled me.

**Reply by merks** [reply]
Yes that’s surprising.

**Comment by ruspl-afed** [comment]
Duplication resolved: Mylyn discovery has been removed, see https://github.com/eclipse-mylyn/org.eclipse.mylyn/issues/196
